scripts/linters/linter_utils.py

```diff
@@ -88,7 +88,7 @@ def get_from_cache(key, factory):
     # In order to allow calls to functions with default arguments to use the
     # same hash as calls which explicitly supply them, we fetch those default
     # values and use them to build the kwargs that func will actually see.
-    arg_names, _, _, defaults = inspect.getargspec(func)
+    arg_names, _, _, defaults, _, _, _ = inspect.getfullargspec(func)
     defaults = defaults if defaults is not None else ()
     default_func_kwargs = dict(
         python_utils.ZIP(arg_names[-len(defaults):], defaults))
```

Comment:
I am not completely familiar with these methods but I looked upon them and tried something like this -
![Screenshot from 2022-01-04 20-03-36 (1)](https://user-images.githubusercontent.com/89375125/148074991-2e65ec76-e664-4b68-89c0-d0d0f34f6cf7.png)

The tests seem to work but again I am no expert. It would be better if someone from a backend team takes a look at it...

Reply:
@IamEzio I took a quick pass and I'm not sure whether we still use the `memoize` decorator anywhere and it looks like a dead code to me, can you please check and remove the `memoize` function in case it's not needed?

Reply:
I think it was expected to be used in [this PR](https://github.com/oppia/oppia/pull/6704/files#diff-c3638bae0e7b24761b2b770244a4c6eaa0da497aef26a06dee081c7668eb6384R370) but never got used.

@brianrodri Do we still need the `memoize` function?

Reply:
@DubeySandeep -- FYI @brianrodri isn't currently around.

Reply:
Thanks for the updates, @seanlip!

@IamEzio Can you please remove the `memoize` function (assuming it's a dead code)?

Reply:
Done